### ISSUE TITLE
Unable to correctly recognize {/} when ifdef/else preprocessor directives are present

### ISSUE THREAD

**Issue by Crazyokd** [miconda/vscode-kamailio-syntax#8]
i found the number of `{` can't be identified correctly in below situation.
```
#!ifdef WITH_AUTH
	if (!ims_www_authenticate("$td")) {
#!else
	if (($var(alg) == "MD5") && (!ims_www_authenticate("$td"))) {
#!endif
```
![image](https://github.com/miconda/vscode-kamailio-syntax/assets/74645100/c0fdf835-4fa7-44fa-a001-a0701213aa23)

whatsoever circumstances, There should be only one valid `{`, but it recognizes it as two, causing an error in bracket matching.

**Comment by miconda** [comment]
I don't think this can be covered without a language server, which is not an easy task. The (not the nicest) variants for now would be:

- move the opening curly brace on the next line:

```
#!ifdef WITH_AUTH
   if( ... )
#!else
   if( ... )
#!end
   {
```

- to have similar ifdef for closing curly braces.

```
#!ifdef WITH_AUTH
   }
#!else
   }
#!end
```

Not having the knowledge nor the time for it, I am going to close this issue soon if nobody else volunteers to develop it.

**Comment by werebear73** [comment]
I have looked at building a language server for this project a couple of times and it makes my head swim.  I haven't given up but it may be awhile.
